data/types/calculator/calculator_test.go

```diff
@@ -1622,6 +1625,13 @@ var _ = Describe("Calculator", func() {
 					pointer.FromString("mg/dl"),
 					func(datum *calculator.Calculator, units *string) { datum.BolusID = nil },
 				),
+				Entry("Invalid Carb Units",
+					pointer.FromString("mg/dl"),
+					func(datum *calculator.Calculator, units *string) {
+						datum.CarbUnits = pointer.FromString("Invalid")
+					},
+					errorsTest.WithPointerSourceAndMeta(structureValidator.ErrorValueStringNotOneOf("Invalid", calculator.CarbUnits()), "/carbUnits", NewMeta()),
+				),
```

Comment:
Also add tests for the 2 valid values.

Reply:
In terms of valid values - it does create valid values (albeit random) in the NewCalculator routine - and that is tested in the first unit test.  We could create one for each valid value - but I do not think that is generally done through the framework.